src/eos_kolibri/eos_kolibri_migrate/main.py

```diff
@@ -7,210 +7,178 @@
 # Authors:
 #  Dylan McCall <dylan@endlessm.com>
 
-import argparse
-import logging
+import click
+
 import os
-import pwd
 import shutil
-import signal
 import subprocess
-import sys
 
+from contextlib import contextmanager
 from pathlib import Path
 
-from . import config
-from .utils import argparse_dir_path, get_backup_path, recursive_chown
+from .. import config
+from ..utils import UserParamType, get_default_user, get_backup_path, recursive_chown
 
-KOLIBRI_DATA_DIR = os.environ.get('KOLIBRI_HOME', config.KOLIBRI_DATA_DIR)
 
-# We ignore the following files from Kolibri's data directory:
-# - .data_version
-# - logs/
-# - server.pid
-# - sessions/
-# - static/
+KOLIBRI_FLATPAK_DATA_PATH = Path(".var/app", config.KOLIBRI_FLATPAK_ID, "data/kolibri")
 
+# These files are sufficient to identify a Kolibri installation
 KOLIBRI_DATA_FILES = (
-    'content',
-    'db.sqlite3',
-    'job_storage.sqlite3',
-    'kolibri_settings.json',
-    'notifications.sqlite3'
+    "content",
+    "db.sqlite3",
 )
 
-def terminate_kolibri_server_process():
-    print("Stopping the Kolibri server...")
-    kolibri_pids = []
-    try:
-        kolibri_pids = subprocess.check_output(['/usr/bin/pgrep', '-x', 'kolibri']).split()
-    except subprocess.CalledProcessError:
-        print("The Kolibri server is not running. Nothing to do.")
-        return
-
-    for pid in kolibri_pids:
-        logging.info("Terminating process with PID {}...".format(pid))
-        os.kill(int(pid), signal.SIGTERM)
 
+def manage_kolibri_service(command):
+    if command not in ("start", "stop"):
+        raise ValueError(f"Invalid service command: {command}")
 
-def manage_kolibri_services(command, types=['socket', 'service']):
-    if command != 'start' and command != 'stop':
-        raise ValueError("Invalid service command: {}".format(command))
+    click.echo(f"Trying to {command} systemd unit: '{config.KOLIBRI_SYSTEMD_SERVICE_NAME}'...")
+    subprocess.check_call(["systemctl", command, config.KOLIBRI_SYSTEMD_SERVICE_NAME])
 
-    for service_type in types:
-        logging.info(
-            "Trying to {} systemd {} unit: '{}'...".format(
-                command, service_type, config.KOLIBRI_SYSTEMD_UNIT_NAME
-            )
-        )
-        subprocess.check_call([
-            '/usr/bin/systemctl',
-            command,
-            '{}.{}'.format(config.KOLIBRI_SYSTEMD_UNIT_NAME, service_type)
-        ])
-
-
-def stop_kolibri_services():
-    print("Stopping Kolibri system services...")
-    manage_kolibri_services('stop')
 
-
-def start_kolibri_services():
-    print("Starting Kolibri system services...")
-    manage_kolibri_services('start', ['socket'])
+@contextmanager
+def stop_kolibri_system_services():
+    click.secho("\nStopping Kolibri...", dim=True)
+    subprocess.run(["killall", "-e", "-w", "kolibri-gnome"])
+    subprocess.run(["killall", "-e", "-w", "kolibri-search-provider"])
+    try:
+        manage_kolibri_service("stop")
```

Comment:
Maybe you could just do and remove `manage_kolibri_service` since the `start` case is no longer used:

```
click.echo(f"Trying to stop systemd unit: '{config.KOLIBRI_SYSTEMD_SERVICE_NAME}'...")
subprocess.check_call(["systemctl", "stop", config.KOLIBRI_SYSTEMD_SERVICE_NAME])
```